features/situation-feedback/elastic/src/main/java/org/opennms/features/situationfeedback/elastic/ElasticHealthCheck.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.features.situationfeedback.elastic;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.opennms.core.health.api.Context;
+import org.opennms.core.health.api.HealthCheck;
+import org.opennms.core.health.api.Response;
+import org.opennms.core.health.api.Status;
+
+import com.google.common.base.Strings;
+
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.core.Ping;
+
+public class ElasticHealthCheck implements HealthCheck {
```

Comment:
There is already an elastic health check implementation. Can't we just re-use that instead of duplicating the code?

Reply:
I wouldn't want another feature to depend on flows-elastic feature - so it would have to be  refactored it out of flows and into the jest feature.
Health check will need 'description' to be parameterized so that if multiple features have different ES config you can easily determine which one is failing. Let me know how strongly you feel about this.

Reply:
If this is duplicated code, the need for re-usage is given.
To me it looks worth tackling.

Reply:
https://github.com/opennms/opennms/blob/823865bf8ec097d693a48479af73d5136472fd84/features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/health/ElasticHealthCheck.java#L54

Reply:
And should be refactored out, maybe as you already suggested, into the jest feature.